cognite_toolkit/_cdf_tk/commands/dump_resource.py

```diff
@@ -557,6 +564,120 @@ def dump_function_code(self, function: Function, folder: Path) -> None:
+
+        Args:
+            app (Streamlit): The Streamlit app whose code is to be dumped.
+            folder (Path): The directory where the app code will be saved.
+            console (Console | None): Optional Rich console for printing warnings.
+        """
+        try:
+            content = self.client.files.download_bytes(external_id=app.external_id)
+        except CogniteAPIError as e:
+            if e.code == 400 and "Files ids not found" in e.message:
+                HighSeverityWarning(
+                    f"The source code for {app.external_id!r} could not be retrieved from CDF."
+                ).print_warning(console=console)
+                return
+            raise
+
+        try:
+            json_content = json.loads(content)
+        except json.JSONDecodeError as e:
+            HighSeverityWarning(
+                f"The JSON content for the Streamlit app {app.external_id!r} is corrupt and could not be extracted. "
+                f"Download file with the same external id manually to remediate. {e!s}"
+            ).print_warning(console=console)
+            return
+
+        app_folder = to_directory_compatible(app.external_id or "unknown_external_id")
+        app_path = folder / app_folder
+        app_path.mkdir(exist_ok=True)
+        if isinstance(json_content.get("requirements"), list):
+            requirements_txt = app_path / "requirements.txt"
+            requirements = json_content["requirements"]
+            if not requirements:
+                HighSeverityWarning(
+                    f"The Streamlit app {app.external_id!r} has a requirements.txt file with no content. Skipping..."
+                ).print_warning()
+            else:
+                requirements_txt.write_text("\n".join(requirements), encoding="utf-8")
+        entry_point = json_content.get("entrypoint")
+        files = json_content.get("files", {})
+        if not isinstance(files, dict) or (not files):
+            HighSeverityWarning(
+                f"The Streamlit app {app.external_id!r} does not have any files to dump. It is likely corrupted."
+            ).print_warning(console=console)
+            return
+        created_files: set[str] = set()
+        for relative_filepath, content in files.items():
+            filepath = app_path / relative_filepath
+            filepath.parent.mkdir(parents=True, exist_ok=True)
+            file_content = content.get("content", {}).get("text", "")
+            if not isinstance(file_content, str):
+                HighSeverityWarning(
+                    f"The Streamlit app {app.external_id!r} has a file {relative_filepath} with invalid content. Skipping..."
+                ).print_warning(console=console)
+                continue
+            safe_write(filepath, file_content, encoding="utf-8")
+            created_files.add(relative_filepath)
+
+        if entry_point and entry_point not in created_files:
```

Comment:
Let's move `entry_point = json_content.get("entrypoint")` from line 654 to here to increase readability